SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -596,6 +599,19 @@ public CreateWarningNoteResponse CreateWarningNote(CreateWarningNoteRequest requ
             return response;
         }
 
+        public List<WarningNote> GetWarningNotes(GetWarningNoteRequest request)
```

Comment:
I suggest making this of type IEnumerable<WarningNote> and then removing ToList from your linq function.

Let the Gateway just get the data and not enforce the structure is is returned in. Also ToList requires iterating over every element to produce the list.

Then in WarningNoteUseCase#ExecuteGet we run another linq function which requires calling .ToList() at the end of the function, essentially returning IEnumerable has 0 impact on the UseCase it is already calling ToList.

Keeps gateway more generic and is also faster.

Reply:
Great suggestion, I have updated the code